power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -260,7 +260,12 @@ struct ComponentToMathCoupling {
 
 // change of update cause topology and param change, or just param change
 struct UpdateChange {
-    bool topo, param;
+    bool topo{};
+    bool param{};
+
+    friend constexpr UpdateChange operator||(UpdateChange const& a, UpdateChange const& b) {
+        return UpdateChange{a.topo || b.topo, a.param || b.param};
```

Comment:
sonarcube suggestion due to variable shadowing

```suggestion
    friend constexpr UpdateChange operator||(UpdateChange const& x, UpdateChange const& y) {
        return UpdateChange{x.topo || y.topo, x.param || y.param};
```